src/overlays/actors/ovl_Bg_Vb_Sima/z_bg_vb_sima.c

```diff
@@ -21,13 +33,129 @@ const ActorInit Bg_Vb_Sima_InitVars = {
     (ActorFunc)BgVbSima_Update,
     (ActorFunc)BgVbSima_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Vb_Sima/BgVbSima_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Vb_Sima/BgVbSima_Destroy.s")
+static InitChainEntry sInitChain[] = { ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP) };
+
+Color_RGBA8_n colorYellow = { 0xFF, 0xFF, 0x00, 0xFF };
+Color_RGBA8_n colorRed = { 0xFF, 0x0A, 0x00, 0xFF };
+
+void BgVbSima_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgVbSima* this = THIS;
+    Actor* actorPtr = &this->dyna.actor;
+    s32 local_c = 0;
+
+    Actor_ProcessInitChain(actorPtr, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 1);
+    DynaPolyInfo_Alloc(&D_06000D68, &local_c);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, actorPtr, local_c);
+}
+
+void BgVbSima_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgVbSima* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgVbSima_SpawnEmber(BossFdParticle* particle, Vec3f* position, Vec3f* velocity, Vec3f* acceleration, f32 scale) {
+    s16 i;
+
+    for (i = 0; i < 180; i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 1;
+            particle->pos = *position;
+            particle->vel = *velocity;
+            particle->accel = *acceleration;
+            particle->scale = scale / 1000.0f;
+            particle->opacity = 0xFF;
+            particle->timer1 = (s16)Math_Rand_ZeroFloat(10.0f);
+            break;
+        }
+    }
+}
+
+void BgVbSima_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgVbSima* this = THIS;
+    BossFd* bossFd = BOSSFD;
+    u32 colPlat;
+    f32 edgeX;
+    f32 edgeZ;
+    s16 i2;
+    s16 i1;
+    Vec3f splashVel;
+    Vec3f splashAcc;
+    Vec3f splashPos;
+    Vec3f emberPos;
+    Vec3f emberVel;
+    Vec3f emberAcc;
+    f32 minus1;
+
+    this->varianceTimer++;
+    if (Flags_GetClear(globalCtx, globalCtx->roomCtx.curRoom.num) == 0) {
```

Comment:
I think `if (!Flags_GetClear(...))` makes more sense since `Flags_GetClear` intends to return a boolean